src/medVtkInria/vtkDataManagement/vtkMetaDataSet.cxx

```diff
@@ -60,9 +60,48 @@ vtkMetaDataSet::vtkMetaDataSet()
   this->PickedCellId     = -1;
   this->Name             = "";
   this->FilePath         = "";
-  this->LookupTable      = NULL;
+  this->LookupTable      = nullptr;
   this->Initialize();
-  
+}
+
+vtkMetaDataSet::vtkMetaDataSet(const vtkMetaDataSet& other)
+{
+  this->DataSet = nullptr;
+  if (other.DataSet)
+  {
+    if (other.DataSet->IsA("vtkPolyData"))
+    {
+        this->DataSet = vtkPolyData::New();
+    }
+    else if (other.DataSet->IsA("vtkUnstructuredGrid"))
+    {
+        this->DataSet = vtkUnstructuredGrid::New();
+    }
+    else if (other.DataSet->IsA("vtkImageData"))
+    {
+        this->DataSet = vtkImageData::New();
+    }
```

Comment:
The above code expects the type of ```DataSet``` to never be anything else, which might change in the future. I think you could replace these lines with the following (not tested):
```this->DataSet = other.DataSet->NewInstance();```

Reply:
Perfect, it's done.